ports/rp2/machine_rtc.c

```diff
@@ -0,0 +1,299 @@
+    mp_uint_t days;
+    days = calendar_to_mjd(t->year, t->month, t->day) - MJD_BASE;
+    return days * 24 * 60 * 60 + t->hour * 3600 + t->min * 60 + t->sec;
+}
+
+
+const mp_obj_type_t machine_rtc_type;
+
+// singleton RTC object
+STATIC machine_rtc_obj_t machine_rtc_obj = {
+    .base = {&machine_rtc_type},
+    .active = false,
+    .period = 0,
+    .alarm = 0,
+};
+
+
+// class machine.RTC(id=0, ...) Constructor
+mp_obj_t machine_rtc_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
+    enum { ARG_id };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_id, MP_ARG_REQUIRED | MP_ARG_OBJ },
+    };
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all_kw_array(n_args, n_kw, all_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    mp_int_t rtc_id = mp_obj_get_int(args[ARG_id].u_obj);
+    if (rtc_id != 0) {
+        mp_raise_msg_varg(&mp_type_ValueError, MP_ERROR_TEXT("RTC(%d) doesn't exist"), rtc_id);
+    }
+    return (mp_obj_t)&machine_rtc_obj;
+}
+
+
+// RTC.init(datetime) method
+// tuple is (year, month, day[, hour[, minute[, second[, microsecond[, tzinfo]]]]])
+STATIC mp_obj_t machine_rtc_init(mp_obj_t self_in, mp_obj_t datetime) {
+    mp_obj_t *items;
+    mp_obj_get_array_fixed_n(datetime, 8, &items);
+    mp_int_t year = mp_obj_get_int(items[0]);
+    mp_int_t month = mp_obj_get_int(items[1]);
+    mp_int_t day = mp_obj_get_int(items[2]);
+    datetime_t t = {
+        .year = year,   // 16 bits
+        .month = month, // 8 bits
+        .day = day,     // 8 bits
+        .dotw = ((calendar_to_mjd(year, month, day) - MJD_BASE) % 7) + MJD_DOTW_BASE,   // 8 bits
+        .hour = mp_obj_get_int(items[3]),  // 8 bits
+        .min = mp_obj_get_int(items[4]),   // 8 bits
+        .sec = mp_obj_get_int(items[5]),   // 8 bits
+    };
+    // ports/rp2/main.c already calls rtc_init(), so we do no init RTC here
+    rtc_set_datetime(&t);
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_2(machine_rtc_init_obj, machine_rtc_init);
+
+
+// RTC.deinit() method
+// sets the date iome as per the MicroPython official documentation
+STATIC mp_obj_t machine_rtc_deinit(mp_obj_t self_in) {
+    datetime_t t = {
+        .year = 2015,
+        .month = 1,
+        .day = 1,
+        .hour = 0,
+        .min = 0,
+        .sec = 0,
+        .dotw = Thursday,
+    };
+    rtc_set_datetime(&t);
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(machine_rtc_deinit_obj, machine_rtc_deinit);
+
+
+// RTC.now() method
+STATIC mp_obj_t machine_rtc_now(mp_obj_t self_in) {
+    datetime_t t;
+    rtc_get_datetime(&t);
+    mp_obj_t tuple[8] = {
+        mp_obj_new_int(t.year),
+        mp_obj_new_int(t.month),
+        mp_obj_new_int(t.day),
+        mp_obj_new_int(t.hour),
+        mp_obj_new_int(t.min),
+        mp_obj_new_int(t.sec),
+        mp_obj_new_int(0),  // usecs
+        mp_const_none       // tzinfo
```

Comment:
The example in the documetation (which sadly is not followed by the ports) puts 0 here instead of None.

Reply:
IMHO, if we are using the standard datetime tuple convention, we should return None, as tzinfo is an object.